public/app/features/dashboard/components/TransformationsEditor/TransformationsEditor.tsx

```diff
@@ -312,29 +377,137 @@ class UnThemedTransformationsEditor extends React.PureComponent<TransformationsE
           </Container>
         )}
         {showPicker ? (
-          <VerticalGroup>
-            <Input
-              aria-label={selectors.components.Transforms.searchInput}
-              value={search ?? ''}
-              autoFocus={!noTransforms}
-              placeholder="Add transformation"
-              onChange={this.onSearchChange}
-              onKeyDown={this.onSearchKeyDown}
-              suffix={suffix}
-            />
-
-            {xforms.map((t) => {
-              return (
-                <TransformationCard
-                  key={t.name}
-                  transform={t}
-                  onClick={() => {
-                    this.onTransformationAdd({ value: t.id });
+          <>
```

Comment:
this is one big-ass component, can we split this up and make it easier to read/test?

Reply:
It was already there, I just extended this part.
I put this part in it s own function, but it is declared inside the existing one, to minimize change.
This is a class component, and there is lot of call to the state and props with `this`. 

Refactoring to a global scope component would require to pass all the state and props, and also add props for all the event handlers used in this part.

Seems like more negative than positive to me.

Maybe we could use some kind of global state and event handlers (with hooks), so we can have a tree of components sharing those without issues.

Seems more like a separate project.

I agree it is bad, but it's already like that and cheap solution would introduce other kind of issues